lib/valueflow.cpp

```diff
@@ -629,7 +629,7 @@ static void setTokenValue(Token* tok,
     if (!parent)
         return;
 
-    if (!isInitList && Token::simpleMatch(parent, ",") && astIsRHS(tok)) {
+    if (!isInitList && Token::simpleMatch(parent, ",") && !parent->isInitComma() && astIsRHS(tok)) {
```

Comment:
Does `isInitComma()` maybe solve the same issue that `isInitList` was added for? Do we need both?
Related: https://trac.cppcheck.net/ticket/11649

Reply:
yes it seems to solve related performance issues. I don't think we need both.. if it sounds ok for you I will remove the isInitList..

Reply:
Did you check that it doesn't re-introduce the previously fixed performance issues? Or is it obvious? I just don't have no understanding what the analysis code is doing. 😀

Reply:
Yes.. I think it would have been good to add a test case. I believe it did not cause a regression but a test would be better.

Reply:
Also a benchmark would be nice. Just to see how much speedup we have gained.

Reply:
A test for 11649 inspired by code in https://github.com/ESA-VirES/WMM/blob/master/EGM9615.h was added in #5380